src/platform/datapath_raw.c

```diff
@@ -485,6 +561,75 @@ CxPlatSocketGetParam(
     return QUIC_STATUS_NOT_SUPPORTED;
 }
 
+CXPLAT_THREAD_CALLBACK(CxPlatRouteResolutionWorkerThread, Context)
+{
+    CXPLAT_ROUTE_RESOLUTION_WORKER* Worker = (CXPLAT_ROUTE_RESOLUTION_WORKER*)Context;
+    CXPLAT_DATAPATH* Datapath = Worker->Datapath;
+
+    while (Worker->Enabled) {
+        CxPlatEventWaitForever(Worker->Ready);
+        if (!CxPlatListIsEmptyNoFence(&Worker->Operations)) {
+            CXPLAT_LIST_ENTRY Operations;
+            CxPlatListInitializeHead(&Operations);
+
+            CxPlatDispatchLockAcquire(&Worker->Lock);
+            if (!CxPlatListIsEmpty(&Worker->Operations)) {
+                CxPlatListMoveItems(&Worker->Operations, &Operations);
+            }
+            CxPlatDispatchLockRelease(&Worker->Lock);
+
+            while (CxPlatListIsEmpty(&Operations)) {
```

Comment:
This lock/list is not a super hot path and doesn't need all the extra complexity here. I think going with a simpler design of lock, pop, unlock, check it cleaner here.